test-runner-frontend/src/Views/ReportUtils.js

```diff
@@ -0,0 +1,86 @@
+import html2canvas from 'html2canvas';
+import jsPdf from 'jspdf';
+
+export const builChartsData = (from) => {
+  const classesAndBenchmarks = {};
+  from?.statistics?.map((testRun, index) => {
+    var benchmarkCassAndMethod = testRun.benchmark.split('.');
+    var benchmarkClass = benchmarkCassAndMethod[1] + '-' + testRun.mode;
+    var benchmarkMethod = benchmarkCassAndMethod[2];
+    var benchmarkData = {
+      jobId: from?.id,
+      benchmarkClass: benchmarkClass,
+      benchmarkMethod: benchmarkMethod,
+      benchmarkScore: testRun.primaryMetric.score,
+      benchmarkError: testRun.primaryMetric.scoreError,
+      mode: testRun.mode,
+      json: testRun,
+    };
+    classesAndBenchmarks[benchmarkClass] ??= [];
+    classesAndBenchmarks[benchmarkClass]?.push(benchmarkData);
+    return classesAndBenchmarks;
+  });
+
+  var allClassesAndBenchmarks = [];
+
+  Object.entries(classesAndBenchmarks).forEach(([key, value]) => {
+    const uniqueIds = new Set();
+    const unique = value.filter((element) => {
+      const isDuplicate = uniqueIds.has(element.benchmarkMethod);
+      uniqueIds.add(element.benchmarkMethod);
+      if (!isDuplicate) {
+        return true;
+      }
+      return false;
+    });
+    allClassesAndBenchmarks[key] = unique;
+  });
+  var sortedByClassNameClassesAndBenchmarks = Object.keys(allClassesAndBenchmarks)
+    ?.sort()
+    ?.reduce(
+      (acc, key) => ({
+        ...acc,
+        [key]: allClassesAndBenchmarks[key],
+      }),
+      {}
```

Comment:
if we want something to be sorted in an order, i would recommend that we don't use an object and prefer an array.

Reply:
Done.